lib/compress/zstdmt_compress.c

```diff
@@ -2076,8 +1773,16 @@ size_t ZSTDMT_compressStream_generic(ZSTDMT_CCtx* mtctx,
             mtctx->inBuff.filled += syncPoint.toLoad;
             forwardInputProgress = syncPoint.toLoad>0;
         }
-        if ((input->pos < input->size) && (endOp == ZSTD_e_end))
-            endOp = ZSTD_e_flush;   /* can't end now : not all input consumed */
+    }
+    if ((input->pos < input->size) && (endOp == ZSTD_e_end)) {
+        /* Can't end yet because the input is not fully consumed.
+            * We are in one of these cases:
+            * - Input buffer is NULL & empty
```

Comment:
note : if `input` buffer is `NULL`,
we should also have : 
`input->pos == 0 == input->size`,
therefore, this scenario shouldn't pass the `if()` test.

Reply:
I should clarify the comment. I mean the `mtctx->inBuff`.